restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java

```diff
@@ -189,10 +192,13 @@
     private SipURI from;
     private javax.servlet.sip.URI to;
     // custom headers for SIP Out https://bitbucket.org/telestax/telscale-restcomm/issue/132/implement-twilio-sip-out
-    private Map<String, String> headers;
+    //headers defined in rcml
+    private Map<String, String> rcmlHeaders;
+    //headers populated by extension to modify existing headers and add new headers
+    private Map<String, ArrayList<String>> extensionHeaders;
```

Comment:
Its quite an effort to design at this stage, we have to expose a new method signature to the `IExtensionCreateCallRequest` interface and we also have to potentially change the `Call` constructor signature etc amongst other things. I would prefer leaving this refactoring for later.